viewing_party/party.py

```diff
@@ -1,23 +1,143 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        return None
+    return {"title": title, "genre": genre, "rating": rating}
 
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == movie_title:
+            add_to_watched(user_data, movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_sum = 0
+    counter = 0
+    try:
+        for movie in user_data["watched"]:
+            rating_sum += movie["rating"]
+            counter += 1
+        return rating_sum / counter
+    except ZeroDivisionError:
+        return 0
 
+def get_most_watched_genre(user_data):
+    # get and store list of all genre values
+    genre_total_list = []
+    for movie in user_data["watched"]:
+        genre_total_list.append(movie["genre"])
+    # create dictionary of most frequent genres
+    genre_frequencies = {}
+    for genre in genre_total_list:
+        try:
+            genre_frequencies[genre] += 1
+        except KeyError: 
+            genre_frequencies[genre] = 1
+    # identify genre with highest frequency count
+    most_popular_genre = None

```

Comment:
A guard clause could be added at the top of this function which might save some code execution and captures the spirit of this assignment of `None`, something like:
```python
    if not user_data["watched"]:
        return None
```